test/gtest/ucp/test_ucp_peer_failure.cc

```diff
@@ -138,12 +142,25 @@ test_ucp_peer_failure::am_callback(void *arg, const void *header,
     return UCS_OK;
 }
 
-void test_ucp_peer_failure::err_cb(void *arg, ucp_ep_h ep, ucs_status_t status) {
+void test_ucp_peer_failure::err_cb(void *arg, ucp_ep_h ep, ucs_status_t status)
+{
     test_ucp_peer_failure *self = reinterpret_cast<test_ucp_peer_failure*>(arg);
     EXPECT_TRUE((UCS_ERR_CONNECTION_RESET == status) ||
                 (UCS_ERR_ENDPOINT_TIMEOUT == status));
     self->m_err_status = status;
     ++self->m_err_count;
+
+    entity *e = self->find_entity_by_ep(ep);
+    if (e == NULL) { /* EP could be revoked from sender's entity */
+        auto iter = self->revoked_eps.find(ep);
+        if (iter != self->revoked_eps.end()) {
+            e = iter->second;
+        }
+    }
+
+    if (e != NULL) {
+        ucp_test::err_handler_cb(e, ep, status);
+    }
```

Comment:
why is it important to call the err_handler_cb of a revoked ep? can't we just ignore it?

Reply:
better to call `ucp_test::err_handler_cb()` for them too.
we could remove `if` and replace with `ASSERT`

Reply:
would be good if we could remove the revoked eps hash

Reply:
> would be good if we could remove the revoked eps hash

ok, removed.
anyway, we're closing UCP EP, so it won't call error handling.